federation/pkg/federatedtypes/adapter.go

```diff
@@ -54,6 +54,11 @@ type FederatedTypeAdapter interface {
 	ClusterWatch(client kubeclientset.Interface, namespace string, options metav1.ListOptions) (watch.Interface, error)
 
 	NewTestObject(namespace string) pkgruntime.Object
+
+	// Individual adapter impl are supposed to provide the hook functions.
+	// Supposed to be nil on those adapters which don't need them (one or both).
+	GetScheduleHook() ScheduleHook
```

Comment:
Thanks for the amazing suggestion. Somehow my thinking got corrupted with the name "hooks" (cause of my earlier engineering experience outside golang, of a hook generally being equal to a function pointer). I find this implementation a lot simpler.
Additionally, more simplification as we also do not need additional IsSchedule() method on the base type adapter as you suggested. Adapter object can be checked for its type runtime. Please check the updated commit.
Conclusively, I find that only one additional method on the wrapper interface is enough (no need of fedUpdateStatus, as per my comment of YAGNI above), which looks a lot simpler where interface definitions are concerned. (I havent made it as 1 method, yet though).
@perotinus need your comments also. What do u think?
The idea is that additional method (only 1) takes the list of objects per cluster (or the list of clusters, with the accessor function), does whatever is needed and returns the objects properly set for distribution into clusters. Can update the fed objects status too, if need be.